lms/djangoapps/verify_student/management/commands/populate_expiry_date.py

```diff
@@ -0,0 +1,96 @@
+"""
+Django admin command to populate expiry_date for approved verifications in SoftwareSecurePhotoVerification
+"""
+import logging
+import time
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from lms.djangoapps.verify_student.models import SoftwareSecurePhotoVerification
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    This command sets the expiry_date for users for which the verification is approved
+
+    The task is performed in batches with maximum number of rows to process given in argument `batch_size`
+    and a sleep time between each batch given by `sleep_time`
+
+    Default values:
+        `batch_size` = 1000 rows
+        `sleep_time` = 10 seconds
+
+    Example usage:
+        $ ./manage.py lms populate_expiry_date --batch_size=1000 --sleep_time=5
+    OR
+        $ ./manage.py lms populate_expiry_date
+    """
+    help = 'Populate expiry_date for approved verifications'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch_size',
+            action='store',
+            dest='batch_size',
+            type=int,
+            default=1000,
+            help='Maximum number of database rows to process. '
+                 'This helps avoid locking the database while updating large amount of data.')
+        parser.add_argument(
+            '--sleep_time',
+            action='store',
+            dest='sleep_time',
+            type=int,
+            default=10,
+            help='Sleep time in seconds between update of batches')
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.sspv = SoftwareSecurePhotoVerification.objects.filter(status='approved',
+                                                                   expiry_date__isnull=True
+                                                                   ).order_by('user_id')
+
+    def handle(self, *args, **options):
+        """
+        Handler for the command
+
+        It creates batches of approved Software Secure Photo Verification depending on the batch_size
+        given as argument. Then for each distinct user in that batch it finds the most recent approved
+        verification and set its expiry_date
+        """
+        batch_size = options['batch_size']
+        sleep_time = options['sleep_time']
+
+        try:
+            max_user_id = self.sspv.last().user_id
```

Comment:
Won't getting `last` cause the entire list to populate?

Reply:
I think I'm unclear on how last works,  In order to get the last item on the list, is a query run, or is a list populated?  One or the other would need to be done in order to know that the given item is the last in the list.  Both cases will have performance issues, one at the DB and the other on memory for the python process running the management command.